runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -1242,6 +1344,8 @@ bool J9::RecognizedCallTransformer::isInlineable(TR::TreeTop* treetop)
          case TR::java_lang_StringCoding_encodeASCII:
          case TR::java_lang_String_encodeASCII:
             return comp()->cg()->getSupportsInlineEncodeASCII();
+         case TR::jdk_internal_util_ArraysSupport_vectorizedMismatch:
```

Comment:
We should not fall back to a common query of supportsArrayCmp() only, Take a look at how String.encodeASCII or other recognized calls are controlled [1], it should do something similar, so other codegens have way to disable that (Selectively disable as well where arraycmp can be supported but the recognizing of this method is disabled)

[1]. https://github.com/eclipse-openj9/openj9/blob/9c8aa3c8d5e6a6aee8a750c85be9649b412ade39/runtime/compiler/z/codegen/J9CodeGenerator.cpp#L121-L124